src/patternfly/components/Drawer/drawer.scss

```diff
@@ -0,0 +1,70 @@
+.pf-c-drawer {
+  --pf-c-drawer__main--MarginRight: calc(var(--pf-c-drawer__panel--Width) * -1);
+  --pf-c-drawer--m-open__main--MarginRight: 0;
+  --pf-c-drawer__main--Transition: margin .25s ease;
+  --pf-c-drawer__panel--ZIndex: var(--pf-global--ZIndex--sm);
+  --pf-c-drawer__panel--Width: 100%;
+  --pf-c-drawer__panel--md--Width: #{pf-size-prem(450px)};
+  --pf-c-drawer__panel--BackgroundColor: var(--pf-global--BackgroundColor--100);
+  --pf-c-drawer__panel--Transition: transform .25s ease;
+  --pf-c-drawer__panel--Transform: translateX(100%);
+  --pf-c-drawer--m-open__panel--Transform: translateX(0);
+  --pf-c-drawer--m-open__panel--BoxShadow: var(--pf-global--BoxShadow--md-left);
+  --pf-c-drawer__panel-body--PaddingTop: var(--pf-global--spacer--md);
+  --pf-c-drawer__panel-body--PaddingRight: var(--pf-global--spacer--md);
+  --pf-c-drawer__panel-body--PaddingBottom: var(--pf-global--spacer--md);
+  --pf-c-drawer__panel-body--PaddingLeft: var(--pf-global--spacer--md);
+  --pf-c-drawer__panel-body--md--PaddingTop: var(--pf-global--spacer--lg);
+  --pf-c-drawer__panel-body--md--PaddingRight: var(--pf-global--spacer--lg);
+  --pf-c-drawer__panel-body--md--PaddingBottom: var(--pf-global--spacer--lg);
+  --pf-c-drawer__panel-body--md--PaddingLeft: var(--pf-global--spacer--lg);
+
+  @media screen and (min-width: $pf-global--breakpoint--md) {
+    --pf-c-drawer__panel--Width: var(--pf-c-drawer__panel--md--Width);
+    --pf-c-drawer__panel-body--PaddingTop: var(--pf-c-drawer__panel-body--md--PaddingTop);
+    --pf-c-drawer__panel-body--PaddingRight: var(--pf-c-drawer__panel-body--md--PaddingRight);
+    --pf-c-drawer__panel-body--PaddingBottom: var(--pf-c-drawer__panel-body--md--PaddingBottom);
+    --pf-c-drawer__panel-body--PaddingLeft: var(--pf-c-drawer__panel-body--md--PaddingLeft);
+  }
+
+  display: flex;
+  overflow-x: hidden;
+
+  &.pf-m-open {
+    --pf-c-drawer__panel--Transform: var(--pf-c-drawer--m-open__panel--Transform);
+
+    &.pf-m-inline {
+      --pf-c-drawer__main--MarginRight: var(--pf-c-drawer--m-open__main--MarginRight);
```

Comment:
How come you override `--pf-c-drawer__main--MarginRight` in `pf-c-drawer` rather than `pf-c-drawer__main`, also I don't see these styles for margin-right working could you explain what they do?

Reply:
one of the benefits of using vars like this is that we can change a subset of children vars on a parent like this. I'm changing the var on `.pf-c-drawer.pf-m-inline` instead of `.pf-c-drawer.pf-m-inline .pf-c-drawer__main`. Progress is a good example of doing this, too - https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/Progress/progress.scss#L50-L52

Reply:
As long as it follows the rules for overriding a variable, then I'm ok with it. To me its harder to follow but maybe thats just me.

Reply:
Also I just realized should the modifier var name change from `--pf-c-drawer--m-open__main--MarginRight` to `--pf-c-drawer--m-open--m-inline__main--MarginRight` ?

Reply:
I just took a glance at other components, looks like we're doing it. We're probably doing both :)

https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/Dropdown/dropdown.scss#L44-L47

https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/Dropdown/dropdown.scss#L55-L58

https://github.com/patternfly/patternfly-next/blob/master/src/patternfly/components/Table/table.scss#L61-L67